feat: improve client interface (#4510)
jina-ai/jina: docs/fundamentals/flow/client.md

client.post('/endpoint')  # Empty
```
### Specifying Host scheme

> To connect to the right `Flow` the client must now the followings parameters:

Reviewer: ```suggestion
To connect to the `Flow` the client has to specify the followings parameters:
```

There is still a verb missing here

Author: my bad again

client.post('/endpoint')  # Empty
```
### Specifying Host scheme

> To connect to the right `Flow` the client must now 

Reviewer: there is some verb missing here?

> * the `protocol`he need to use to communicate with the `Flow

Reviewer: I think the client is 'it' not he

* the `host` and the `port`on which the Flow is exposed
* if he needs to use `tls` encryption

> You can precise these parameters by passing a valid scheme to the `host` argument:

Reviewer: ```suggestion
You can define these parameters by passing a valid URI scheme as part of the `host` argument:
```

```python
from jina import Client

Client(host='https://my.awesome.flow:1234')
Client(host='wss://my.awesome.flow:1234')
Client(host='grpcs://my.awesome.flow:1234')
```

> You can as well pass each relevant parameters as a keyword argument:

Reviewer: ```suggestion
You can as well pass each relevant parameter as a keyword argument:
```

the following example is equivalent to the one above
```python
Client(host='my.awesome.flow', port=1234, protocol='http', tls=True)
Client(host='my.awesome.flow', port=1234, protocol='websocket', tls=True)
Client(host='my.awesome.flow', port=1234, protocol='grpc', tls=True)
```

You can use a mixe of both as well:

```python
Client(host='https://my.awesome.flow', port=1234)
Client(host='my.awesome.flow:1234', protocol='http', tls=True)
```

````{admonition} Caution
:class: caution

> You can't precise these parameters by keyword arugment and by host schem (You can't have two sources of truth)

Reviewer: ```suggestion
You can't define these parameters by keyword arugment and by host scheme (You can't have two sources of truth)
```
